### PR TITLE
Feat/character UI

### LINKED ISSUE #1165: new-game-ui : Character switching + Plaza + Upgrade/skins/options
Clicking characters is static currently needs to be fix

Screen Shot 2022-02-10 at 6 04 58 PM

image

Skins
image

Options
image

Change Name and Change Trait

https://xd.adobe.com/view/61961669-fb51-4607-8e31-030fbca6400f-34c5/screen/1ff07b6d-a9b8-47f7-9ec1-4b0a50818722/

### PR DESCRIPTION
**Pull request by gabo2192**
### All Submissions

* [ ] Can you post a screenshot of your changes (if applicable)?

### New Feature Submissions

* [X ] Does this relate to an existing issue, or has it been talked about prior to being done (if a major change)?
Relates to #1165

### Notes

Please note, your code must pass all tests and lint checks before it can be merged.

### PR Description
This is the new UI Layout
Closes #1165

### Testing

Has the code been tested, or does it need double checking?

### CODE REVIEW

File: frontend/src/views/Character.vue [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+  }
+});
+</script>
+<style scoped lang="scss">
+
+.background-image {
+  background-image: url('../assets/artwork-fantasy-art-ruins.png') ;
+  background-size: cover;
+  background-repeat: no-repeat;
+  background-position: top right;
```
Comment (Sellie96) [inline frontend/src/views/Character.vue:230]: Can this have a min height for full view height. Can see the problem in the garrison tab.

Also can you apply a black opaque overlay over the image like in some of the other pages.

File: frontend/src/components/smart/Character.vue [hunk 2]
```diff
@@ -0,0 +1,657 @@
...
+        </div>
+        <div class="col cell">
+          <span class="main-font cell-title">{{$t(`Character.reputation`)}}</span>
+          <span class="cell-value">{{$t(`quests.reputationTier.${ReputationTier[getReputationLevel(reputation)]}`)}}</span>
+        </div>
+        <div class="col cell">
+          <span class="main-font cell-title">{{$t(`Character.element`)}}</span>
+          <p class="alt-text cell-value">
+            <span :class="characterTrait.toLowerCase() + '-icon circle-element'"></span>
+            {{characterTrait}}
```
Comment (Sellie96) [inline frontend/src/components/smart/Character.vue:47]: Can all of the above have seperate classes applied to distinguish the data from the value. 


e.g Element   "Earth" coloured green
Level   "1" the one being different

Just generally highlight the data from the text

File: frontend/src/components/smart/CharacterTabs/UpgradeTab.vue [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+          </div>
+        </div>
+        <div class="col-2 character-text d-flex">
+          <input id="powerAmount" type="number" :value="powerAmount" @change="handleInput" />
+          <button class="mx-1 px-2"  @click="handleMax">Max</button>
+        </div>
+      </div>
+      <button class="upgrade-character-button mt-5" @click="onUpgradeConfirm">
+        <span>{{$t('Character.upgrade')}}</span><br/>{{powerAmount/10}} {{$t(`Character.soul`)}}
+      </button>
```
[the remark this answers is not in the document]
Comment (Sellie96) [reply frontend/src/components/smart/CharacterTabs/UpgradeTab.vue:35]: I see it's on the Garrison tab but probably best we keep it on the main screen too

File: frontend/src/components/smart/Character.vue [hunk 4]
```diff
@@ -0,0 +1,657 @@
...
+    getCharacterArt,
+    RequiredXp,
+    openTransferModal(){
+      (this.$refs['character-transfer-modal'] as BModal).show();
+    },
+    openChangeTrait(){
+      (this.$refs['character-change-trait-modal'] as BModal).show();
+    },
+    async openChangeNameModal() {
+      this.haveRename = await this.fetchTotalRenameTags();
```
Comment (Sellie96) [inline frontend/src/components/smart/Character.vue:376]: This is returning an error

**Comment by Sellie96** [comment]
Don't see anyway to burn characters, need to leave in the old option if not updating as part of this PR

File: frontend/src/components/smart/Character.vue (hunk 4 above)
Comment (gabo2192) [reply frontend/src/components/smart/Character.vue:381]: fixed

File: frontend/src/views/Character.vue (hunk 1 above)
Comment (gabo2192) [reply frontend/src/views/Character.vue:464]: fixed

File: frontend/src/components/smart/CharacterTabs/UpgradeTab.vue [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+            <p class="text-center character-text">{{powerAmount}}/{{remainingPowerLimit}} Power</p>
+          </div>
+        </div>
+        <div class="col-2 character-text d-flex">
+          <input id="powerAmount" type="number" :value="powerAmount" @change="handleInput" />
+          <button class="mx-1 px-2"  @click="handleMax">Max</button>
+        </div>
+      </div>
+      <button class="upgrade-character-button mt-5" @click="onUpgradeConfirm">
+        <span>{{$t('Character.upgrade')}}</span><br/>{{powerAmount/10}} {{$t(`Character.soul`)}}
```
[the remark this answers is not in the document]
Comment (gabo2192) [reply frontend/src/components/smart/CharacterTabs/UpgradeTab.vue:34]: updated

File: frontend/src/components/smart/CharacterTabs/UpgradeTab.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (gabo2192) [reply frontend/src/components/smart/CharacterTabs/UpgradeTab.vue:35]: fixed

File: frontend/src/views/Character.vue [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+  },
+  async mounted(){
+    this.checkStorage();
+  },
+  async created(){
+    this.mintPriceDecreasePerHour = new BN(await this.fetchMintCharacterPriceDecreasePerSecond()).div(new BN(10).pow(18)).multipliedBy(60*60).toFixed(6);
+    this.mintCharacterPriceIncrease = new BN(await this.fetchCharacterMintIncreasePrice()).div(new BN(10).pow(18)).toFixed(6);
+    this.mintCharacterMinPrice = new BN(await this.fetchMintCharacterMinPrice()).div(new BN(10).pow(18)).toFixed(4);
+    this.updateMintCharacterFee();
+    this.updateInterval = setInterval(async () => { await this.updateMintCharacterFee(); }, 2000);
```
[the remark this answers is not in the document]
Comment (gabo2192) [reply frontend/src/views/Character.vue]: This logic was inherit, I updated it so it only updates when a new character is minted

File: frontend/src/components/smart/Character.vue (hunk 2 above)
Comment (gabo2192) [reply frontend/src/components/smart/Character.vue:51]: I tried adding a slightly dark background and using primary color for the text so we don't fail on visibility

**Comment by gabo2192** [comment]
> 

This is fixed already.

**Comment by gabo2192** [comment]
> 

readded

**Comment by gabo2192** [comment]
> Bug: Select a character in the garison => Click back to plaza => Everything NaN.
> 
> I'm guessing you need to select when to the first character when leaving. But whatever solution you want is fine

This is fixed already, when they get back to character page it checks if there's no characters selected, if none it selects the first one.

**Comment by gabo2192** [comment]
> Don't see anyway to burn characters, need to leave in the old option if not updating as part of this PR

This got readded.

File: frontend/src/views/Character.vue [hunk 7]
```diff
@@ -0,0 +1,522 @@
+
+<template>
+  <div class="p-5 mb-5 background-image">
```
Comment (Sellie96) [inline frontend/src/views/Character.vue:3]: The margin bottom here needs removing. It's applying to the background image stopping it being the full screen

**Comment by Sellie96** [comment]
Can you do a double check of the Mobile UI please. There are a few issues that cause the screen to be weird. Notably the skin section + Buttons for minting and burning

**Comment by gabo2192** [comment]
>

**Comment by gabo2192** [comment]
> Can you do a double check of the Mobile UI please. There are a few issues that cause the screen to be weird. Notably the skin section + Buttons for minting and burning

I did some arrangenments for Mobile UI, let me know if anything else is needed :)

**Review by Sellie96 — APPROVED** [review]

**Pull request merged**